File: javascript/easy/reverseLinkedList.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    if(head == null) return null;
    let current = head, runner = head.next, temp = null;
    head.next = null;
    while(runner != null) {
        temp = runner.next;
        runner.next = current;
        current = runner;
        runner = temp;
    }
    return current;
};

File: ruby/easy/reverse_linked_list.rb
# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val)
#         @val = val
#         @next = nil
#     end
# end

# @param {ListNode} head
# @return {ListNode}
def reverse_list(head)
    return nil unless head
    current, runner = head, head.next
    head.next = nil
    while runner
        temp = runner.next
        runner.next = current
        current = runner and runner = temp
    end
    return current
end